features/behat-steps.feature

```diff
@@ -0,0 +1,741 @@
+    And a file1.txt file:
+      """
+      content1
+      """
+    And a file2.txt file:
+      """
+      content2
+      """
+    Then the {RUN_DIR} directory should contain:
+      """
+      file1.txt
+      """
+    And the {RUN_DIR} directory should contain:
+      """
+      file2.txt
+      """
+
+  Scenario: Test "I run the previous command again" step
+    When I run `echo "test"`
+    Then STDOUT should contain:
+      """
+      test
+      """
+
+    When I run the previous command again
+    Then STDOUT should contain:
+      """
+      test
+      """
+
+  Scenario: Test variable replacement in commands
+    When I run `echo "myvalue"`
+    Then save STDOUT as {TEST_VAR}
+
+    When I run `echo "Value is: {TEST_VAR}"`
+    Then STDOUT should contain:
+      """
+      myvalue
+      """
+
+  Scenario: Test STDOUT strictly be
+    When I run `echo "exact"`
+    Then STDOUT should strictly be:
+      """
+      exact
+      """
+
+  Scenario: Test file strictly contain
+    Given an empty directory
+    And a strict.txt file:
+      """
+      exact content
+      """
+    Then the strict.txt file should strictly contain:
+      """
+      exact content
+      """
+
+  @require-wp
+  Scenario: Test WP installation steps
+    Given a WP installation
+    When I run `wp core version`
+    Then STDOUT should not be empty
+    And the return code should be 0
+
+  @require-wp
+  Scenario: Test WP files and wp-config.php steps
+    Given an empty directory
+    And WP files
+    Then the wp-settings.php file should exist
+
+    Given wp-config.php
+    Then the wp-config.php file should exist
+    And the wp-config.php file should contain:
+      """
+      DB_NAME
+      """
+
+  @require-wp
+  Scenario: Test WP database step
+    Given a WP installation
+    And a database
+    When I run `wp db check`
+    Then the return code should be 0
+
+  @require-wp
+  Scenario: Test WP installation in subdirectory
+    Given a WP installation in 'subdir'
+    When I run `wp core version` from 'subdir'
+    Then STDOUT should not be empty
+    And the return code should be 0
+
+  @require-wp
+  Scenario: Test plugin installation step
+    Given a WP installation
+    And these installed and active plugins:
+      """
+      hello
+      """
+    When I run `wp plugin list --status=active --field=name`
+    Then STDOUT should contain:
+      """
+      hello
+      """
+
+  @require-wp
+  Scenario: Test version string comparison
+    Given a WP installation
+    When I run `wp core version`
+    Then STDOUT should be a version string >= 4.0
+
+  @require-wp
+  Scenario: Test STDOUT as table containing rows
+    Given a WP installation
+    When I run `wp option list --fields=option_name --format=table --orderby=option_name`
+    Then STDOUT should be a table containing rows:
+      | option_name |
+
+  @require-wp
+  Scenario: Test JSON output
+    Given a WP installation
+    When I run `wp option get siteurl --format=json`
```

Comment:
`wp option` is not  available in this repo at the moment.

So alternatively, here, instead of running a WP-CLI command, you could do something like ```When I run `echo <json string>` ``` and then assert the output of that.